tests/Unit/Domain/Test_ElementId.cpp

```diff
@@ -108,6 +112,7 @@ void test_element_id() {
 
   // Test pup operations:
   test_serialization(element_one);
+  CHECK(PUP::size(element_one) == 3 * sizeof(int));
```

Comment:
For my understanding, why is this being tested? I understood that the size of an ElementId was the condition on whether it could serve as a charm index. But is it actually the size of the pup'd object?

Reply:
it wasn't before the change to PUPbytes; then the size PUP saw was 6 ints as it made two short ints plus an int for each dimension...

Reply:
I'm trying to understand the underlying requirements... Is the requirement for using ElementId as a charm index that (a) `sizeof(ElementId) == 3 * sizeof(int)`, or (b) `sizeof(serialized(ElementId)) == 3 * sizeof(int)`?

If (a), then I don't understand why (b) is being tested, and a comment above this line that explains the significance of (b) would be helpful.

If (b), then this PR should also update the documentation of ElementId/SegmentId to clarify the relevant size... because the documentation currently implies it is the unPUP'd size that matters. For example, I suggest changing lines 57-58 of ElementId.hpp to the following:
"- `ElementId` must serialize into a buffer no larger than the size of 3 `int`s, i.e. `sizeof(serialized<ElementId>(element_id)) == 3 * sizeof(int)`" (the LHS here might be incorrect code, but hopefully you understand what I am trying to suggest). Same for SegmentId.hpp.

If (a) and (b), then I suggest changing lines 57-58 of ElementId.hpp to
"- `ElementId` (and its serialization) must not be ...."

Reply:
the requirement is (a); I will drop the tests of the PUP::size...

Reply:
Thanks for clarifying 🙂